### ISSUE TITLE
[Solidity] Build failure due to system linker

### ISSUE THREAD

**Issue by bshastry** [google/oss-fuzz#10237]
For the proto fuzzers, the following failure happens. This is unrelated to upstream project.

```
/usr/bin/ld.gold: internal error in read_header_prolog, at ../../gold/dwarf_reader.cc:1678
clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
```

Any idea why?

**Comment by jonathanmetzman** [comment]
Nope. I don't think we made any changes recently and other projects do not appear affected.

**Comment by DavidKorczynski** [comment]
I think we need to adjust similar to, which seems to have missed solidity https://github.com/google/oss-fuzz/commit/a455e651de867eb7aed4c351b424d1d9538f4e50

**Comment by bshastry** [comment]
Thank you so much. Here's my understanding of the breaking change and the fix. Please correct me if I'm wrong

1. LPM downgraded from c++14 to c++17. This somehow required protobuf libraries to be linked together and not in arbitrary order as was the case before.

2. Upstream projects that make use of protobuf fuzzers hence need to respect this linker grouping.

Solidity makes use of c++17. Would this be a problem?

**Comment by DavidKorczynski** [comment]
> Solidity makes use of c++17. Would this be a problem?

@vitalybuka could you assist here?

**Comment by bshastry** [comment]
> > I'll give it a try and update here
> 
> It's probably easier for you @bshastry as it looks to be deeper in the solidity build systems (https://github.com/ethereum/solidity/blob/a00a2117c5aceb9c77e4e970196f3208ed1ea25c/test/tools/ossfuzz/CMakeLists.txt#L67-L70 ?) in either case, the above commit and issue should clarify why the protobuf fuzzers started failing and how to fix it.

Sadly, the suggested fix does not work. FWIW, I'm linking an address sanitized c++17 build of solidity. Error is same as in original post.

**Comment by bshastry** [comment]
The only "fix" I've been able to come up with so far to get the protobuf fuzzers to build is to use the LPM commit prior to the breaking change (specifically this commit: https://github.com/google/libprotobuf-mutator/commit/212a7be1eb08e7f9c79732d2aab9b2097085d936)

**Comment by bshastry** [comment]
Could you please let me know how to fix this cleanly. For now, I've created a PR to use a hard-coded commit (prior to the LP upgrade, see https://github.com/google/oss-fuzz/pull/10301) that does not fail the solidity build. Adding the `-Wl --start-group, -lprotobuf --Wl --end-group` to the linker options does not work unfortunately.
